build.rs

```diff
@@ -28,15 +30,14 @@ fn is_docs_rs() -> bool {
 }
 
 fn is_sidecar_ffi_enabled() -> bool {
-    // Note: arm64 is disabled for now as cross-compilation is not yet supported in rust2go
-    // and it's reported rust2go build does not work on arm64 MacOS
-    if cfg!(target_arch = "x86_64") {
+    // Note: arm64 is disabled on MacOS for now as it's reported rust2go build does not work there
+    if cfg!(target_arch = "aarch64") && cfg!(target_os = "macos") {
```

Comment:
Alternative:
```suggestion
    if cfg!(all(target_arch = "aarch64", target_os = "macos")) {
```
I have no preference, though.